src/javascript/app/Stores/Modules/Trading/trade-store.js

```diff
@@ -279,10 +279,21 @@ export default class TradeStore extends BaseStore {
                         ...this.proposal_info[type],
                         buy_price: response.buy.buy_price,
                     };
+                    const {
+                        contract_id,
+                        longcode,
+                        start_time,
+                    } = response.buy;
                     // toggle smartcharts to contract mode
-                    const contract_id = getPropertyValue(response, ['buy', 'contract_id']);
                     if (contract_id) {
-                        this.root_store.modules.contract.onMount(contract_id);
+                        // NOTE: changing chart granularity and chart_type has to be done in a different render cycle
+                        // so we have to set chart granularity to zero, and change the chart_type to 'mountain' first,
+                        // and then set the chart view to the start_time
+                        this.root_store.modules.smart_chart.switchToContractMode();
```

Comment:
Maybe can combine `this.root_store.modules` into a `const`?
`const module = this.root_store.modules;`
`modules.smart_chart.switchToContractMode();`
`...`